### PR TITLE
Update text-stream-search

### LINKED ISSUE #7: Allow for regex in wait method
The ObservableProcess.wait method should be able to wait for strings that include a regex for cases such as waiting for a string that includes a version number that could change from time to time.

### PR DESCRIPTION
**Pull request by hugobho**
@kevgo 

resolves #7

### CODE REVIEW

File: features/auto-detect-path.feature [hunk 1]
```diff
@@ -10,7 +10,7 @@ Feature: Auto-detecting the path to the executable
 
   Scenario: relative path given
     Given I run the global command "node -h"
-    Then it prints "Usage: node [options]"
+    Then it prints "Usage: node \[options\]"
```
Comment (kevgo) [inline features/auto-detect-path.feature]: Okay, I didn't consider that using a regex has such side effects on normal text. Let's go with your original idea of having two separate steps:

```cucumber
Then it prints /regex text/
Then it prints "normal text"
```

Then you can revert this change.

**Comment by hugobho** [comment]
@kevgo reverted.
